### PR TITLE
Adding type hints to feature_permutation.py

### PR DESCRIPTION
**Pull request by cykustcc**
Add type hints for the following two files:
```
captum/attr/_core/feature_permutation.py
tests/attr/test_feature_permutation.py
```

Test:
```
python -m unittest -v tests.attr.test_feature_permutation OK
./scripts/run_mypy.sh shows no errors:
```

### CODE REVIEW

**Comment by NarineK** [comment]
@cykustcc thank you for working on this PR. I have just renamed this files and changed the class names for consistency reasons. Make sure that you rebase from master.

**Comment by cykustcc** [comment]
Hi @NarineK, I have rebased to master and resolve the name conflict. PTAL. Thanks!

File: tests/attr/test_feature_permutation.py [hunk 1]
```diff
@@ -99,15 +99,15 @@ def forward_func(x):
...
         batch_size = 20
         inp1_size = (5, 2)
         inp2_size = (5, 3)
 
         labels = torch.randn(batch_size)
 
-        def forward_func(*x):
-            y = 0
+        def forward_func(*x: Tensor) -> Tensor:
+            y = torch.zeros(x[0].shape[0:2])
```
Comment (vivekmig) [inline tests/attr/test_feature_permutation.py]: nit: It might be more concise to just have torch.zeros_like(x[0]) here.

**Comment by cykustcc** [comment]
@NarineK, I double checked the versions of flake8 and black on my devgpu. They seem to be the latest. And they give no compliant on my modification. Could you recommend me which version of them should be used?
```
flake8 --version
3.7.9 (mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.1) CPython 3.7.4 on Linux
black --version
black, version 19.10b0
```

File: captum/attr/_core/feature_permutation.py [hunk 2]
```diff
@@ -19,18 +23,30 @@ def _permute_feature(x, feature_mask):
...
     def attribute(
         self,
-        inputs,
-        target=None,
-        additional_forward_args=None,
-        feature_mask=None,
-        ablations_per_eval=1,
-    ):
+        inputs: TensorOrTupleOfTensors,
+        baselines: Optional[
```
[the remark this answers is not in the document]
Comment (cykustcc) [reply captum/attr/_core/feature_permutation.py]: If I don't add baselines argument, mypy will give the following type checking error:
```
captum/attr/_core/feature_permutation.py:34: error: Signature of "attribute" incompatible with supertype "FeatureAblation"
Found 1 error in 1 file (checked 44 source files)
```

File: tests/attr/test_feature_permutation.py (hunk 1 above)
Comment (cykustcc) [reply tests/attr/test_feature_permutation.py]: Replace it with ```torch.zeros_like(x[0])``` will give the following Runtime Error in testing:
```
RuntimeError: The size of tensor a (2) must match the size of tensor b (5) at non-singleton dimension 2
```

**Comment by vivekmig** [comment]
Hi @cykustcc, the new versions won't revert the changes made to other files, but they likely won't make the same changes / would accept the existing code, so we'd preferably like to keep them to what they were before and only modify the relevant files. You would need to reset to current master for all the files other than feature_permutation and test_ feature_permutation, this would revert your changes on those files. Thanks!

File: captum/attr/_core/feature_permutation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (vivekmig) [reply captum/attr/_core/feature_permutation.py]: Good point! You can just add the comment "# type: ignore" to the end of the line, and it'll suppress the warning. This error is caused by the child class not having a matching signature to the parent (doesn't take a baseline), and we may look into changing the class hierarchy in the future, but for now, you can just add the ignore flag. You can also add a short comment above explaining the reason.

**Comment by vivekmig** [comment]
Looks great! You can Import to Phabricator now.

**Pull request closed without merging**